### ISSUE TITLE
unable to pull docker container

### ISSUE THREAD

**Issue by hughcross** [nf-core/eager#337]
Hello,

I am trying to run eager using Singularity, but whenever I try, it fails with this error:

> Unable to pull docker://nfcore/eager:2.0.7: While searching for mksquashfs: exec: "mksquashfs": executable file not found in $PATH

I have tried this with the test, my own data, and just trying to pull the Singularity image separately:

singularity pull --name eager-2.0.7.img docker://nfcore/eager:2.0.7

Please let me know if there is something I am doing wrong, or if this missing exec is a known issue. Thank you

**Comment by jfy133** [comment]
Hey Hugh, 

It looks like you're missing a dependency for singularity, like you said.

Please see the following: https://github.com/sylabs/singularity/issues/2542#issuecomment-454145861

I would also recommend trying use the `-r dev` version of EAGER. 2.0.7 is very old now and we have a rebuilt version in the beta version coming soon(tm). Nextflow should pull the correct singularity image for you if singularity is working probably, so you shouldn't need to pull it manually.

**Comment by hughcross** [comment]
Thanks James,
I was able to fix the singularity issue and pull the image. However, everything I try to run terminates with errors, even the test set. I also tried with some trial data that worked fine with version 2.0.6 (using conda install). The error I keep seeing is:

ERROR ~ Error executing process > 'output_documentation'

I ran the test set as recommended:
nextflow run nf-core/eager -profile singularity,test --pairedEnd

**Comment by jfy133** [comment]
Hi Hugh,

Glad you got singularity to work. 2.0.7 is possibly broken because of some changes to the way the docker images are built (whihc singularity then pulls).

Could you instead try with the current beta version (with `-r dev`). This is a partly rewritten version of the pipeline (substantially so, which is why it's not yet released).

i.e.
```
nextflow pull nf-core/eager -r dev
nextflow run nf-core/eager -profile singularity,test --pairedEnd -r dev
```

If it doesn't work, please could you send me the `.nextflow.log` file associated with the run.

**Comment by hughcross** [comment]
Hi,

Thanks for your continued help. The dev version threw an error right away. log file attached.
[nextflow.log](https://github.com/nf-core/eager/files/4136416/nextflow.log)

**Comment by jfy133** [comment]
Hi Hugh,

Thanks for the log file. That helps a lot.

Line 27: 

```
Jan-31 10:53:22.181 [main] WARN  nextflow.script.ScriptRunner - Nextflow version 19.01.0 does not match workflow required version: >=19.10.0 -- Execution will continue, but things may break!
```

Then crashes at line 308 

```
Jan-31 10:53:23.740 [main] DEBUG nextflow.Session - Session aborted -- Cause: Unknown process directive: `path`
```

path is a new operator ('function') which isn't in nextflow 19.01.0 (which is is quite an old version now).

I would recommend upgrading nextflow to 19.10.0 and trying again.

But it's useful to know it doesn't kill the pipeline with a useful error message. I will therefore add a note to the installation and running documentation that EAGER2 needs a later version. Thanks for reporting!

Let me know if you still have problems once upgrading!

**Comment by apeltzer** [comment]
Thanks for taking care of this @jfy133 ! Hugh, let us know if there are some more issues once you've upgraded your Nextflow version to 19.10.0 or later. You can do that by running `nextflow self-update`, which should download the latest stable release of Nextflow for you 👍

**Comment by hughcross** [comment]
Thanks for all your help. unfortunately, I am still getting errors. I tried with the test and also my own data, which had worked with eager 2.0.6. Logfiles are attached.
[nextflow.log](https://github.com/nf-core/eager/files/4150167/nextflow.log)
[nextflow2.log](https://github.com/nf-core/eager/files/4150170/nextflow2.log)

**Comment by apeltzer** [comment]
Sad to hear! 

Are you running Singularity on your cluster? Has it been upgraded recently? 

This looks like it can't find the files by the pipeline unfortunately, which sometimes points a bit towards how Singularity mounts host paths of the machine / host you're running the pipeline on. Can you do this here:

```
singularity shell work/singularity/path_to_eager.img
cd /working/beavers/dev_tests/work
ls -lah
```

(Censor the output if this is somewhat internal data - it only matters to see whether Singularity is capable to access the paths it has to have access to - looks a bit similar to this one here: https://nf-co.re/usage/troubleshooting#cannot-find-input-files-when-using-singularity)

**Comment by jfy133** [comment]
@hughcross I'm with Alex on this one. It very much looks like for whatever reason singularity is not being able to access the singularity container.

Plesae try what he suggested. From your log files, the image should be in `/working/beavers/dev_tests/work/singularity/nfcore-eager-dev.img`

In the meantime, if you are able to set up `conda`, and run `-profile test,conda` that would be a good way of testing whether this is a singularity or EAGER error. It's slower but often more reliable as it is less complicated to set up.

**Comment by jfy133** [comment]
FYI another user has had the same problem, again with what appears to be a fresh singularity install. We should make sure that if either @hughcross or the other user solves this, we record this in the FAQ.

It's definitely a singularity set-up issue, as I can run it fresh on my laptop with no issue, and conda also works for the other user.

**Comment by hughcross** [comment]
I am sorry it has taken me so long to get back to this. I was able to get the singularity version working on the test data and one of my own data sets. I had to re-install singularity (still version 3.2.0), then added the paths to the working folders to the singularity.conf file:
bind path = /working
bind path = /home

as suggested by Alex, from link above. I will test on some other data sets as well, and at some point try to update Singularity. This is on our local server (on which I have root access). I have asked our national server IT to upgrade Nextflow, and once they do, I will try on that server. 

Thanks for all your help. I will post back if I run into any other issues.
